### ISSUE TITLE
Memory consumption issue after upgrade to Centos 8

### ISSUE THREAD

**Issue by Rohlik** [grobian/carbon-c-relay#434]
Hello there 👋🏼,
after what we hit on #432 we decided to upgrade our production server to Centos 8 and enable native SSL in Carbon-C-relay, but since then we have problem with memory consumption. Number of received metrics should be same as before upgrade to CO8 but memory usage is much more higher and it increase over time until carbon-c-relay gets kill by OOM.

On that virtual server **only** carbon-c-relay is running.
This behavior is happening on all of our Centos 8 servers with carbon-c-relay native SSL enabled.

Histogram from one of our server shows state before upgrade and after (7.6. afternoom):
![image](https://user-images.githubusercontent.com/2051154/121851782-d56e7200-ccee-11eb-8e1e-dd14552c676f.png)

Process:
```
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                                                                                                                               
 161082 carbon-+  20   0  236.2g  30.3g   5252 S 155.4  77.2   3413:06 carbon-c-relay
```
CMD:
`/usr/bin/carbon-c-relay -f /etc/carbon-c-relay.conf -c -_:#^ -b 2500 -q 7500000 -l /mnt/log/carbon-c-relay/carbon-c-relay.log
`

Kernel (uname -r): `4.18.0-240.22.1.el8_3.x86_64`
OS: `CentOS Linux release 8.3`
Carbon-c-relay: `carbon-c-relay-3.7.2`

**Comment by grobian** [comment]
feels like it leaks a handle or something, thanks

**Comment by grobian** [comment]
Do you frequently reload the configuration by chance?  That's the only leak I've been able to find sofar.

**Comment by Rohlik** [comment]
@grobian Well, we are using Puppet to delivery changes every hour, but we don't reload configuration if there is no update 🤔 .
Anyways, we had same Puppet even on Centos 7 and there wasn't any issues with memory leaks as you can see on graph.

**Comment by grobian** [comment]
This looks a lot like issue #439, but there CentOS 7 is used, and also there I seem to be completely unable to reproduce the problem :(

**Comment by berrfred** [comment]
I suggest you give a try to the latest release 3.7.3, that solved our SSL memory leak issue on CentOS 7.

**Comment by Rohlik** [comment]
Well i wasn't successful as @grobian recently bump version of autoconf to `2.71` and latest version available for Centos 8 is `2.69` 😞 , so I cannot build RPM for our environment.

**Comment by grobian** [comment]
hmmmm...

Did you use the tar that I uploaded to the release?  It shouldn't require autotools.  If you want to use the github zip, you'll have to use ./configure --disable-maintainer-mode and probably touch some files that would be generated by bison and flex.

**Comment by Rohlik** [comment]
Thanks @grobian , you are right. It's not needed for tar archive. Thank you. We will proceed with update carbon-c-relay.

**Comment by Rohlik** [comment]
Problem looks solved after upgrade to latest version `2.7.3` in our DEV env.
![image](https://user-images.githubusercontent.com/2051154/143866832-fb87fb0c-b6bf-4082-a503-6d194f99afcf.png)
